src/mlir/logical_result.rs

```diff
@@ -0,0 +1,82 @@
+use mlir_sys::MlirLogicalResult;
+
+/// [`LogicalResult`] wraps the `llvm::LogicalResult` class, an unowned fragment of a string.
+pub struct LogicalResult {
+    raw: MlirLogicalResult,
+}
+
+impl LogicalResult {
+    /// Constructs a [`LogicalResult`] from a provided bool value representing whether the result is a success.
+    ///
+    /// # Arguments
+    /// * `is_success` - The bool value.
+    ///
+    /// # Returns
+    /// Returns a new [`LogicalResult`] instance.
+    pub const fn success() -> Self {
+        Self {
+            raw: MlirLogicalResult { value: 1 },
+        }
```

Comment:
The spec defines an optional `isSuccess` arg to  `LogicalResult::success`, should we implement our functions closer to that spec?

I am fine with the current implementation because it forces to more intuitive uses of the function, but just wondering if there are cases where the argument would be necessary. 

Reference: https://mlir.llvm.org/doxygen/structmlir_1_1LogicalResult.html#a61f94d5a1103576957132f9be7b781de